chain/jsonrpc/src/lib.rs

```diff
@@ -1295,6 +1327,7 @@ pub fn start_http(
     client_addr: Addr<ClientActor>,
     view_client_addr: Addr<ViewClientActor>,
     #[cfg(feature = "adversarial")] peer_manager_addr: Addr<PeerManagerActor>,
+    #[cfg(feature = "adversarial")] ibf_routing_pool: Addr<RoutingTableActor>,
```

Comment:
Why do we need to pass `ibf_routing_pool` here? We already have peer_manager's address, and peer_manager has ibf pool already. Can we communicate with peer manager for adversarial needs?

It doesn't feel right that we need to thread ibf_pool, which is a private impl detail, up to top-level start_http call. #[cfg()] function parameters are also odd to work with. 


I wonder...  should we perhaps instead make existing `AdversarialControls` a singleton instead, accessible everywhere? I think for such kind of code globals are better than threading impl details everywhere. Eg, tikvs failpoints are global-based: https://docs.rs/fail/0.4.0/fail/. Although it probably still better not to make it a literal static variable (which doesn't jive well with threading), but just pass an `Arc<AdversarialControlls>` everywhere we need them?

Reply:
This is needed for tests. After this change we will have copy of routing table in both `RoutingTableActor` and `PeerManagerActor`. I modified a python test, that uses jsonrpc, to request routing table  from both `RoutingTableActor`, `PeerManagerActor`, there was actually a bug, where those two wouldn't be in sync.

Unfortunately, actors `handle` api is synchronous, so there is no way for `PeerManager` to return result from `RoutingTableActor` in `handle` method. The only way is to modify `jsonrpc` server to ask `RoutingTableActor` directly.